internal/provider/resource_tfe_workspace_execution_mode.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		return fmt.Errorf("error attaching agent pool ID %s to workspace ID on CREATE %s: %w", agentPoolID, workspaceID, err)
+	}
+
+	d.SetId(workspace.ID)
+	// d.Set() will update the resource state file for execution_mode and agent pool ID
+	d.Set("execution_mode", workspace.ExecutionMode)
+	d.Set("agent_pool_id", workspace.AgentPool.ID)
+
+	return resourceTFEWorkspaceExecutionModeRead(d, meta)
+}
+
+func resourceTFEWorkspaceExecutionModeRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	log.Printf("[DEBUG] Read configuration: %s", d.Id())
+	workspace, err := config.Client.Workspaces.ReadByID(ctx, d.Id())
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Workspace %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading configuration of workspace %s: %w", d.Id(), err)
+	}
+
+	d.Set("workspace_id", workspace.ID)
+
+	var poolID string
+	if workspace.AgentPool != nil {
+		poolID = workspace.AgentPool.ID
+	}
+	d.Set("agent_pool_id", poolID)
+
+	return nil
+}
+
+func resourceTFEWorkspaceExecutionModeUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	workspaceID := d.Get("workspace_id").(string)
+
+	options := tfe.WorkspaceUpdateOptions{}
+
+	if d.HasChange("execution_mode") {
+		options.ExecutionMode = tfe.String(d.Get("execution_mode").(string))
+	}
+
+	if d.HasChange("agent_pool_id") {
+		options.AgentPoolID = tfe.String(d.Get("agent_pool_id").(string))
+	}
+
+	log.Printf("[DEBUG] Update attachment on workspace with agent pool ID: %s", d.Get("agent_pool_id"))
+	workspace, err := config.Client.Workspaces.UpdateByID(ctx, workspaceID, options)
+	if err != nil {
+		return fmt.Errorf("error attaching agent pool ID %s to workspace ID %s: %w", d.Get("agent_pool_id"), workspaceID, err)
+	}
+
+	d.SetId(workspace.ID)
+	// d.Set() will update state file on tfe_workspace for execution_mode and agent pool ID
+	d.Set("execution_mode", workspace.ExecutionMode)
+	d.Set("agent_pool_id", workspace.AgentPool.ID)
+
+	return resourceTFEWorkspaceExecutionModeRead(d, meta)
+}
+
+func resourceTFEWorkspaceExecutionModeDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	poolID := d.Get("agent_pool_id").(string)
+	workspaceID := d.Get("workspace_id").(string)
+
+	log.Printf("[DEBUG] Detach agent pool %s from workspace %s", poolID, workspaceID)
+
+	_, err := config.Client.Workspaces.ReadByID(ctx, workspaceID)
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Workspace %s no longer exists", workspaceID)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading configuration of workspace %s: %w", workspaceID, err)
+	}
+
+	_, errs := config.Client.Workspaces.UpdateByID(ctx, workspaceID, tfe.WorkspaceUpdateOptions{
+		AgentPoolID:   tfe.String(""),
+		ExecutionMode: tfe.String("remote"),
```

Comment:
Now that there is an organization default exection mode, I wonder if we should set it to _that_

Reply:
Not sure. That PR isn't merged yet and we will likely merge this one, first. @brandonc